core-bundle/src/Messenger/AutoFallbackWorker.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Messenger;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Output\NullOutput;
+use Symfony\Component\EventDispatcher\Attribute\AsEventListener;
+use Symfony\Component\HttpKernel\Event\TerminateEvent;
+use Symfony\Component\Messenger\Command\ConsumeMessagesCommand;
+use Symfony\Component\Messenger\Event\WorkerRunningEvent;
+use Symfony\Component\Messenger\Event\WorkerStartedEvent;
+use Symfony\Component\Messenger\Event\WorkerStoppedEvent;
+
+/**
+ * This service accepts an array of Symfony Messenger transports and automatically
+ * detects which of those have real workers running. In case they do not, it falls
+ * back to a kernel.terminate worker to ensure, messages are always processed, no
+ * matter whether a real worker is running or not.
+ *
+ * Detecting works by using the WorkerStartedEvent and WorkerRunningEvent which
+ * sets/updates a cache items and allowing a grace period.
+ *
+ * In any case, this provides advantages as it allows us to always use a queue and
+ * postpone processes to at least kernel.terminate in case there are no real
+ * workers available!
+ */
+class AutoFallbackWorker
+{
+    private const GRACE_PERIOD = 'PT60S'; // If no real worker "pings" us within this grace period, we run our fallback worker
```

Comment:
Do we still need to increase this because of the reasons outlined in https://github.com/contao/contao/issues/7251#issue-2325195841 ?

Reply:
Ah, I see, the `You cannot call ack()` error is gone, so we can be more agressive with the grace period I think. I’d still prefer setting it to something like 10 minutes or so, to reduce the chance of unintentionally switching to the fallback handling.

Reply:
That would mean:

- for the case that your cron fails, it would take 10 minutes for the fallback handling to process the messages
- for the case that a task takes very long it would enable the fallback handling unintentionally only after 10 minutes

right?

Reply:
Exactl.y

Reply:
Adjusted :)